### PR TITLE
[web] Migrate Flutter Web to JS static interop - 6.

### PR DESCRIPTION
**Pull request by joshualitt**
This is CL 6 in a series of CLs to migrate Flutter Web to the new JS static interop API.

This CL migrates more of canvaskit_api.dart(about 40% through that file). I stopped after SkPaint because it was a little more involved.

### CODE REVIEW

**Comment by joshualitt** [comment]
@srujzs ptal, this one is a bit more involved because of the replacement of a generative constructor with a factory constructor, but not by much.

File: lib/web_ui/lib/src/engine/canvaskit/canvaskit_api.dart [hunk 1]
```diff
@@ -904,12 +919,17 @@ class SkMaskFilterNamespace {
...
+// with `new`.
+// Also in Dart you can't write this:
 //
-//     external SkPaint SkPaint();
+//     external SkPaint SkPaint.create();
 @JS('window.flutterCanvasKit.Paint')
+@staticInterop
 class SkPaint {
-  external SkPaint();
+  external factory SkPaint.create();
```
Comment (srujzs) [inline lib/web_ui/lib/src/engine/canvaskit/canvaskit_api.dart:929]: Any reason we want a named factory? Named and unnamed factories get lowered to the same code anyways here.

Comment (joshualitt) [reply lib/web_ui/lib/src/engine/canvaskit/canvaskit_api.dart]: soooo much better, must smaller diff now. ptal.

**Comment by srujzs** [comment]
LGTM

**Comment by Hixie** [comment]
test-exempt: code refactor with no semantic change

**Pull request merged**